### DISCUSSION TITLE
Not able to convert key from String to Key format

### DISCUSSION THREAD

**Discussion by SPrabhudesai** [krzyzanowskim/ObjectivePGP#215] [General]
From mail getting key in string format so how I convert that into Key format 

The error:- Cannot assign value of type 'String.SubSequence' (aka 'Substring') to type 'Key?'

**Comment by krzyzanowskim** [comment]
https://github.com/krzyzanowskim/ObjectivePGP#read-keys-private-or-public

**Comment by SPrabhudesai** [comment]
Where is conversion ?

**Comment by krzyzanowskim** [comment]
I'm not sure I understand what is the "Key format" ?

**Comment by SPrabhudesai** [comment]
i am send private key with mail send with string when i Decrypt message i use the key which i send with mail but for Decryption the key is key format and my key is in String format so how i convert the key from String to key format?

**Comment by krzyzanowskim** [comment]
I can guess you're talking about Armored message. If to then you can read it using this: https://github.com/krzyzanowskim/ObjectivePGP#read-keys-private-or-public and you get PGPKey instances that you can use further

**Comment by SPrabhudesai** [comment]
No i sending key with PGP encrypted message to receiver Now receiver at time of sending its in string format but at time Decryption i use same key which i sending with mail.So that key getting with String format now i want to covert that string into key format because Decryption mail.
 So how can i convert the key from String format into key format?

**Comment by krzyzanowskim** [comment]
1. https://github.com/krzyzanowskim/ObjectivePGP#read-keys-private-or-public
2. https://github.com/krzyzanowskim/ObjectivePGP#export-keys-private-or-public
